common/include/pcl/correspondence.h

```diff
@@ -59,30 +59,24 @@ namespace pcl
   struct Correspondence
   {
     /** \brief Index of the query (source) point. */
-    int index_query;
+    int index_query = 0;
     /** \brief Index of the matching (target) point. Set to -1 if no correspondence found. */
-    int index_match;
+    int index_match = -1;
     /** \brief Distance between the corresponding points, or the weight denoting the confidence in correspondence estimation */
     union
     {
-      float distance;
+      float distance = std::numeric_limits<float>::max();
       float weight;
     };
-    
-    /** \brief Standard constructor. 
-      * Sets \ref index_query to 0, \ref index_match to -1, and \ref distance to FLT_MAX.
-      */
-    inline Correspondence () : index_query (0), index_match (-1), 
```

Comment:
You need to default this instead of removing. The default ctor will not be created due to the user provided one

Reply:
Right, true. Didn't see the other constructor